src/main/java/com/redhat/devtools/intellij/knative/kn/ServiceDeserializer.java

```diff
@@ -0,0 +1,83 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.redhat.devtools.intellij.knative.kn;
+
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.deser.std.StdNodeBasedDeserializer;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ServiceDeserializer extends StdNodeBasedDeserializer<Service> {
+
+    public ServiceDeserializer() {
+        super(Service.class);
+    }
+
+    @Override
+    public Service convert(JsonNode root, DeserializationContext deserializationContext) {
+        String name = root.get("metadata").get("name").asText();
+        JsonNode jsonStatus = root.get("status");
+
+        ServiceStatus status = convertToStatus(jsonStatus);
+
+        return new Service(name, status);
+    }
+
+    private ServiceStatus convertToStatus(JsonNode statusNode) {
+        String url = statusNode.get("url").asText();
+        int observedGeneration = statusNode.get("observedGeneration").asInt();
+        String latestReadyRevisionName = statusNode.get("latestReadyRevisionName").asText();
```

Comment:
Better to add a check in these cases. If you try to add a service which will fail (e.g add a not-existing image and deploy) the service will end up in the state below and this will result in a NPE
![image](https://user-images.githubusercontent.com/49404737/108822285-2848e000-75bf-11eb-9450-cc3da55dd6a7.png)